KS-07-05-password-generator-update

DenisGubernatorov opened the pull request:
1) Генерация пароля с использованием диапазонов 'A'..'Z','a'..'z' и '0'..'9' 2) Пользователю не позволяется ввести длину пароля меньше чем 6 символов

DenisGubernatorov committed:
KS-07-05-password-generator-update
1) Генерация пароля с использованием диапазонов 'A'..'Z','a'..'z' и '0'..'9'
2) Пользователю не позволяется ввести длину пароля меньше чем 6 символов

ievetrovdev commented on src/main/kotlin/lesson7/lesson7_task5.kt:
```diff
@@ -1,2 +1,17 @@
 package lesson7
 
+const val MIN_PASSWORD_LENGTH = 6
+val range = ('A'..'Z').toList() + ('a'..'z').toList() + (0..9).toList()
```
Лучше объявить эту переменную внутри `fun main()`

ievetrovdev commented on src/main/kotlin/lesson7/lesson7_task5.kt:
```diff
@@ -1,2 +1,17 @@
+    }
+
+    var generatedPassword = ""
+    for (i in 0..passwordLength) generatedPassword += range.random()
```
Ошибка в диапазоне цикла. Пароль генерируется на один символ больше, чем требуется.

DenisGubernatorov committed:
KS-07-05-password-generator-update
1) Исправлена ошибка, что длина пароля больше на один символ
2) range сделана локальной переменной в main, а не свойством класса

DenisGubernatorov commented:
Поправил

ievetrovdev reviewed (changes requested):
Окей код работает.  Стоит еще конечную строку с паролем перемешать. Сделать это можно, используя extension `shuffle().`
Но обрати внимание с какими типами работает   `shuffle().`. Тебе необходимо чтобы твой пароль состоял из трёх составных частей, у тебя это прописано в `val range` , но немного не верно указано  **toList()** . Тебе нужно поработать с типом  `Char`  `CharRange`

DenisGubernatorov committed:
KS-07-05-password-generator-update
1) Исправлено формирование интервала с доступными символами. Теперь это список char, а не объектов реализующих Comparable и Serializable
2) Окончательный пароль формируется путем перемешивание строки, состоящей из символов полученных случайным образом из допустимого диапазона

DenisGubernatorov commented:
С range  понял. Тип объектов в list  был действительно не Char.  Но вот дополнительное перемешивание результирующего пароля не лишнее? Из-за псевдослучайноси? Но ведь по дефолту генерация зависит от параметров системы. Вроде как время в этом учавствует.

ievetrovdev reviewed (approved):
В обычных условиях генерация пароля с помощью стандартных методов случайных чисел не должна зависеть от параметров системы. Что касается дополнительного перемешивания сгенерированного пароля, то лишним точно не будет.
